packages/extension-core/src/handlers/Extension.ts

```diff
@@ -78,33 +77,22 @@ export default class Extension extends ExtensionHandler {
       return store
     })
 
-    awaitKeyringLoaded().then(() => {
-      // Watches keyring to do things that depend on type of accounts added
-      keyring.accounts.subject.subscribe(async (addresses) => {
-        const sites = await stores.sites.get()
-
-        Object.entries(sites)
-          .filter(([, site]) => site.connectAllSubstrate)
-          .forEach(async ([url, autoAddSite]) => {
-            const newAddresses = Object.values(addresses)
-              .filter(
-                ({ json: { meta } }) =>
-                  isTalismanHostname(autoAddSite.url) ||
-                  ![AccountType.Watched, AccountType.Dcent].includes(meta.origin as AccountType),
-              )
-              .filter(({ json: { address } }) => !autoAddSite.addresses?.includes(address))
-              .map(({ json: { address } }) => address)
-
-            autoAddSite.addresses = [...(autoAddSite.addresses || []), ...newAddresses]
-            await stores.sites.updateSite(url, autoAddSite)
-          })
-      })
+    keyringStore.accounts$.subscribe(async (accounts) => {
+      const sites = await stores.sites.get()
 
-      this.stores.app.observable.subscribe(({ onboarded }) => {
-        if (onboarded === "TRUE") {
-          this.checkSpiritKeyOwnership()
```

Comment:
After removing the call here, we aren't calling `this.checkSpiritKeyOwnership` anywhere anymore. I think we might need to add it back somewhere?